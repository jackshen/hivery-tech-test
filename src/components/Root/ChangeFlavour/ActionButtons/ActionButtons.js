import React from "react";
import styled from "styled-components";

import Button from "#root/components/shared/Button";

const ButtonWrapper = styled.div`
  :not(:last-child) {
    margin-right: 0.5rem;
  }
`;

const Wrapper = styled.div`
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-end;
  grid-area: action-buttons;
  padding: 0.75rem 1rem;
`;

const ActionButtons = ({ canCancel, canSave, onCancel: pushCancel }) => {
  return (
    <Wrapper>
      <ButtonWrapper>
        <Button disabled={!canCancel} onClick={() => pushCancel()}>
          Cancel
        </Button>
      </ButtonWrapper>
      <ButtonWrapper>
        <Button color="primary" disabled={!canSave}>
          Save
        </Button>
      </ButtonWrapper>
    </Wrapper>
  );
};

export default ActionButtons;
