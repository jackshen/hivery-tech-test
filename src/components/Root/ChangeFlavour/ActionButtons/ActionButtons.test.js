import { fireEvent } from "@testing-library/react";
import React from "react";

import wrapAll from "#test/wrapAll";

import ActionButtons from "./ActionButtons";

describe("ActionButtons", () => {
  test("onCancel works properly", () => {
    const handleCancelMock = jest.fn();

    const { getByText } = wrapAll(<ActionButtons canCancel onCancel={handleCancelMock} />);

    fireEvent.click(getByText("Cancel"));

    expect(handleCancelMock).toHaveBeenCalledTimes(1);
  });

  test("onCancel doesn't trigger if canCancel is false", () => {
    const handleCancelMock = jest.fn();

    const { getByText } = wrapAll(<ActionButtons canCancel={false} onCancel={handleCancelMock} />);

    fireEvent.click(getByText("Cancel"));

    expect(handleCancelMock).not.toHaveBeenCalled();
  });

  // the Save button isn't actually hooked up to anything, so we'll just make do with a snapshot test for now
  test("canSave disables the Save button", () => {
    const { container } = wrapAll(<ActionButtons canCancel canSave={false} />);

    expect(container).toMatchSnapshot();
  });
});
