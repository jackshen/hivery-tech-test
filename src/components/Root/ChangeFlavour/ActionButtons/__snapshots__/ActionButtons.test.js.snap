// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButtons canSave disables the Save button 1`] = `
.c2 {
  background-color: white;
  border: 0.0625em solid #d8d8d8;
  border-radius: 0.25rem;
  font-family: inherit;
  font-size: 1.1rem;
  padding: 0.5em 1.125em;
  -webkit-transition: background-color 0.25s,border-color 0.25s,box-shadow 0.25s,color 0.25s;
  transition: background-color 0.25s,border-color 0.25s,box-shadow 0.25s,color 0.25s;
}

.c2:focus {
  box-shadow: 0 0 0 2px white,0 0 3px 3px #1890ff;
  outline: none;
}

.c2:hover {
  background-color: #f2f2f2;
  cursor: pointer;
}

.c2:disabled {
  background-color: #eeeeee;
  border-color: #cccccc;
  color: #999999;
  cursor: not-allowed;
}

.c3 {
  background-color: white;
  border: 0.0625em solid #d8d8d8;
  border-radius: 0.25rem;
  font-family: inherit;
  font-size: 1.1rem;
  padding: 0.5em 1.125em;
  -webkit-transition: background-color 0.25s,border-color 0.25s,box-shadow 0.25s,color 0.25s;
  transition: background-color 0.25s,border-color 0.25s,box-shadow 0.25s,color 0.25s;
  background-color: #1890ff;
  border-color: #1890ff;
  color: white;
}

.c3:focus {
  box-shadow: 0 0 0 2px white,0 0 3px 3px #1890ff;
  outline: none;
}

.c3:hover {
  background-color: #f2f2f2;
  cursor: pointer;
}

.c3:disabled {
  background-color: #eeeeee;
  border-color: #cccccc;
  color: #999999;
  cursor: not-allowed;
}

.c3:not(:disabled):hover {
  background-color: #0084fe;
  cursor: pointer;
}

.c1:not(:last-child) {
  margin-right: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  grid-area: action-buttons;
  padding: 0.75rem 1rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="c2"
      >
        Cancel
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        class="c3"
        color="primary"
        disabled=""
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
